tools/kinesis_ingestor/timestream_kinesis_data_gen.py

```diff
@@ -228,9 +228,19 @@ def create_record(measure_name, measure_value, value_type, timestamp, time_unit)
     }
 
 
-def send_records_to_kinesis(all_dimensions, kinesis_client, stream_name):
+def send_records_to_kinesis(all_dimensions, kinesis_client, stream_name, sleep_time, percent_late, late_time):
     while True:
```

Comment:
When we will finish this while loop?

Reply:
This is by design and part of the original code. The utility is designed to run until the user ^C out of it. It is simply generating records for Kinesis.